### DISCUSSION TITLE
Welcome to DStep Discussions!

### DISCUSSION THREAD

**Discussion by jacob-carlborg** [jacob-carlborg/dstep#266] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

To start a new topic/discussion, please use this link: https://github.com/jacob-carlborg/dstep/discussions/new. For questions, please use the [Q&A](https://github.com/jacob-carlborg/dstep/discussions/categories/q-a) category. This category supports marking discussions as answered.

**Comment by cpiker** [comment]


**Reply by jacob-carlborg** [reply]
> Thanks for the invite to discuss dstep topics here. This is the first time I've attempted to be more engaged with the D community and I have to say everyone I've messaged in the last two weeks has been extraordinarily helpful and, just plain nice. It's a great precedent to set and I hope it helps build a bigger D community.

That's great to hear.

> On the topic of module names, the reason that I used deimos as the top level module name for my CDF wrappers is due to this diagram from the D wiki article of the same name.

Ok, I see. It's not a big deal and as you mentioned, it predates Dub.

> Later as more knowledge is gained a full up D implementation of spice is implemented. Without the deimos "namespace", how should we distinguish between the two? One of them would end up with the name spiced or something cute like that and the other would just be spice. The end user now has to investigate the two to see which better fits their needs.

Sure, that can happen. It can also happen that someone wants to create another set of bindings for the same library. We already have bindings for WinAPI in druntime but there's still a separate Dub project with bindings for WinAPI.

I don't think there's one simple answer for all this. I would not necessarily name the Dub package the same as the C project it contains bindings for. It depends on the name of the C library. I think there are many C libraries that have quite poor names. They name the library after what it does and prepend `lib` to it. Like `libhttp`, `libyaml` and so on. Having a Dub project just called `http` is a bit too generic for me. It's even worse with the Foundation framework on macOS. Having a Dub project just named `foundation`, I don't know if anyone would understand the connection to the Foundation framework.

Of course, everyone if free to do as they wish, I just pointed out the standard D conventions 😃.

**Comment by jacob-carlborg** [comment]
@cpiker BTW, I'm new to GitHub discussions, but I think the intention is that you would create new discussion/topic and not reusing an existing one like this. This was just an auto-generated introduction/welcoming discussion.

**Reply by jacob-carlborg** [reply]
> You might consider modifying the default instructions with a hint to start new threads

Done.

> I don't know if you can move messages to new threads, but please consider doing so for this one if you can.

No, it doesn't look like that is supported.

**Reply by jacob-carlborg** [reply]
Sure.
